server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -76,13 +76,18 @@ public final long getUsageCount() {
     @Override
     public abstract List<Route> routes();
 
+    private static final Set<String> ALWAYS_SUPPORTED = Set.of("format", "Accept", "filter_path", "pretty", "human");
+
     @Override
     public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
         // check if the query has any parameters that are not in the supported set (if declared)
         Set<String> supported = supportedQueryParameters();
```

Comment:
Could these be done with a default implementation of `supportedQueryParameters()`? Implementations will need to call super, but that seems better to have what is returned by that method be correct, rather than partial and some magic set added later.

Reply:
We'll be looking at this as a whole later on, this PR is just so specifying `pretty=true` doesn't break rest queries that specify capability params

Reply:
Ok, so once we've added all implementations, we can switch to relying on super? Let's just make sure we call super to begin with?